src/style-spec/expression/types/resolved_image.js

```diff
@@ -1,18 +1,25 @@
 // @flow
 
+export type ResolvedImageOptions = {
+    name: string,
+    available: boolean
+};
+
 export default class ResolvedImage {
     name: string;
+    available: boolean;
 
-    constructor(name: string) {
-        this.name = name;
+    constructor(options: ResolvedImageOptions) {
+        this.name = options.name;
+        this.available = options.available;
     }
 
     toString(): string {
         return this.name;
     }
 
-    static fromString(imageName: string): ResolvedImage {
-        return new ResolvedImage(imageName);
+    static fromString(options: ResolvedImageOptions): ResolvedImage {
```

Comment:
Hmm, that's a good point. It's used in other places where it might be less straightforward to use the constructor though (https://github.com/mapbox/mapbox-gl-js/blob/master/src/style-spec/expression/definitions/coercion.js#L105). I'm inclined to leave this as is though because it follows the pattern of other expression types and this type and operator will likely be expanded significantly in the future as we add more image manipulation operators.